lazy.md

```diff
@@ -13,3 +13,623 @@ author:
+can live side by side: not all functionality has to be implemented
+by the same coroutine task. The objective for this proposal is to
+select a set of features which is provides a coroutine task suitable
+for most uses. It may also be reasonable to provide some variations
+as different names. A future revision of the standard or third party
+libraries can also provide additional variations.
+
+# Design
+
+This section discusses various design options for achieving the
+listed objectives. Most of the designs are independent of each other
+and can be left out if the consensus is that it shouldn't be used
+for whatever reason.
+
+## Template Declaration for `lazy`
+
+**TODO** turn into text
+
+- Coroutines can `co_return` a value; if no argument is given or
+    no `co_return` is used, the argument type is void
+- The return type `T` can become a value `set_value_t(T)` or
+    `set_value_t()` completion
+- The return type cannot be deduced => it needs to be explicit on
+    `lazy<T>`
+- For various features it is beneficial to have an additional,
+    defaulted template argument for a context (see below for details):
+    - provide an initial environment for child senders
+    - define the default scheduler affinity
+    - define whether coroutine is `noexcept`
+    - define allocator awareness
+    - etc.
+- The default context should be explicity named and provide "safe"
+    defaults
+    - inheriting from the default context can be used to override
+        defaults
+- Having different types for `lazy<T, C>` isn't a problem because
+    these entities aren't stored in containers: instead they are
+    maintained via something like `counting_scope` which starts the
+    work and cleans up upon completion
+
+## `lazy` constructors and assignments
+
+**TODO**
+
+- use guidance from [P1056](https://wg21.link/p1056) and [P2506](https://wg21.link/p2506)
+
+## Result Type For `co_await`
+
+**TODO** turn into text
+
+- status quo based on `execution::as_awaitable(sender)`:
+    - Exactly one `set_value_t(T...)` completion
+    - `set_value_t()` => `void`
+    - `set_value_t(T)` => `decay_t<T>`
+    - `set_value_t(T0, T1, ...)` => `tuple<decay_t<T>...>`
+- `co_await just_stopped()` could indicate cancellation; supported
+   by stdexec.
+- `co_await just_error(error)` would be reasonable for completions
+    always indicating failure; supported by stdexec.
+- the result type is already inconsistent instead of always
+    `tuple<decay_t<T>...>` => support multiple `set_value_t`
+    completions and produce a `variant<tuple<T...>...>`
+- the last option is different from `co_await into_variant<sender>`
+    as it treats `set_stopped_t()` and `set_error_t(error)` differently
+- it may be reasonable to detect specific shapes of completion
+    signatures and turn them into something more coroutine friendly like
+    an `optional<T>` or an `expected<T, E>`; alternatively, support
+    corresponding algorithms like `into_optional`, `into_expected`
```

Comment:
I am default-suspicious of this design direction, but I'll listen to arguments. I don't think the default coroutine type should perform very much magic beyond the surprise-reducing scheduler affinity.